xbmc/video/VideoInfoScanner.cpp

```diff
@@ -121,36 +121,45 @@ namespace VIDEO
       // result in unexpected behaviour.
       m_bCanInterrupt = false;
 
-      bool bCancelled = false;
-      while (!bCancelled && !m_pathsToScan.empty())
+
+      if (m_handle)
+        m_handle->SetTitle("Listing media files ...");
+
+      // List all media files of all 'm_pathsToScan'
+      //   INFO: m_pathsToScan may contain path with different library content, so we store their properties into a dedicated struct.
+      std::vector<SourcePathContent> _pathsContent;
+      CFileItemList remoteDirectoryItems[m_pathsToScan.size()];
+      int nbRemoteDirectoryItems = 0;
+
+      for (const auto& rootPathToScan : m_pathsToScan)
       {
-        /*
-         * A copy of the directory path is used because the path supplied is
-         * immediately removed from the m_pathsToScan set in DoScan(). If the
-         * reference points to the entry in the set a null reference error
-         * occurs.
-         */
-        std::string directory = *m_pathsToScan.begin();
-        if (m_bStop)
+        // Load root path properties from scraper settings:
+        bool foundDirectly = false;
+        bool bSkip = false;
+        SScanSettings rootPathScraperScanSettings;
+        ScraperPtr info = m_database.GetScraperForPath(rootPathToScan, rootPathScraperScanSettings, foundDirectly);
+        CONTENT_TYPE contentType_AsDefinedInScraper = info ? info->Content() : CONTENT_NONE;
+
+        // List and store medias:
+        if( isPathAllowedForScan(rootPathToScan, rootPathScraperScanSettings, contentType_AsDefinedInScraper))
         {
-          bCancelled = true;
+          loadRemoteItems(rootPathToScan, remoteDirectoryItems[nbRemoteDirectoryItems++], rootPathScraperScanSettings, contentType_AsDefinedInScraper);
+          _pathsContent.push_back({rootPathToScan, rootPathScraperScanSettings, contentType_AsDefinedInScraper, remoteDirectoryItems[nbRemoteDirectoryItems-1]});
+
+          //We use the 'path' of CFileItemList to store its root-path-to-scan
+          remoteDirectoryItems[nbRemoteDirectoryItems-1].SetPath(rootPathToScan);
         }
-        else if (!CDirectory::Exists(directory))
-        {
-          /*
-           * Note that this will skip clean (if m_bClean is enabled) if the directory really
-           * doesn't exist rather than a NAS being switched off.  A manual clean from settings
-           * will still pick up and remove it though.
-           */
-          CLog::Log(LOGWARNING, "{} directory '{}' does not exist - skipping scan{}.", __FUNCTION__,
-                    CURL::GetRedacted(directory), m_bClean ? " and clean" : "");
-          m_pathsToScan.erase(m_pathsToScan.begin());
+
+        // Support scan cancellation
+        if (m_bStop) {
```

Comment:
coding guidelines: put `{` on next line. (https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#31-braces) - here and everywhere else in your PR.